infra/aws/terraform/registry-k8s-io-prod/cloudfront.tf

```diff
@@ -0,0 +1,114 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+resource "aws_cloudfront_distribution" "cloudfront_distribution" {
+  provider = aws.networking
+
+  enabled         = true
+  is_ipv6_enabled = true
+
+  origin_group {
+    origin_id = "registry-k8s-io-prod-image-layers"
+
+    failover_criteria {
+      status_codes = [403, 404, 500, 502]
+    }
+
+
+    member {
+      origin_id = "registry-k8s-io-prod-us-east-2"
+    }
+
+    member {
+      origin_id = "registry-k8s-io-prod-eu-west-1"
+    }
+  }
+
+  origin {
+    domain_name = "prod-registry-k8s-io-us-east-2.s3.us-east-2.amazonaws.com"
+    origin_id   = "registry-k8s-io-prod-us-east-2"
+
+    origin_shield {
+      enabled              = true
+      origin_shield_region = "us-east-2"
+    }
+  }
+
+  origin {
+    domain_name = "prod-registry-k8s-io-eu-west-1.s3.eu-west-1.amazonaws.com"
+    origin_id   = "registry-k8s-io-prod-eu-west-1"
+
+    origin_shield {
+      enabled              = true
+      origin_shield_region = "eu-west-1"
+    }
```

Comment:
Why do we need to use origin shield in this case? My understanding is that this is useful when you have a non-AWS origin, but given that this is a S3 bucket, I don't think it provides many benefits while it adds to the cost.

Reply:
Apparently this is not a significant cost so we can leave it enabled